### PR TITLE
CSS-10140 Implemented linters and CI/CD pipelines.

### PR DESCRIPTION
**Pull request by mertalpt**
Changelog:
- Overhaul the choice of linters and `tox` actions.
- Add `.github` directory with action definitions and issue templates.
- Iterate on the code until the linters stop complaining. Changes are mostly on docstrings and trimming line lengths.

Made draft until new actions can be tested.

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+name: Tests
+
+on:
+  pull_request:
+
+jobs:
+  unit-tests:
+    uses: canonical/operator-workflows/.github/workflows/test.yaml@cea2ac306b4f4c1475d73b1a4c766d62e5b1c8a9
```
Comment (kelkawi-a) [inline .github/workflows/test.yaml:8]: Try running the latest commit of `test.yaml`. We fixed the commit at the time due to instabilities with the workflow at the time.

**Review by kelkawi-a — APPROVED** [review]

File: src/charm.py [hunk 2]
```diff
@@ -92,11 +91,21 @@ def get_pebble_layer(service: services.AbstractService, context: services.Servic
 
 
 class DatahubK8SOperatorCharm(TypedCharmBase[CharmConfig]):
-    """Charm the application."""
+    """Charm the application.
+
+    Attributes:
+        _state: Used to store persistent data between invocations.
+        config_type: CLass used to store the config.
```
Comment (AmberCharitos) [inline src/charm.py:98]: nitpick: Accidental capital `L` in `Class`

File: src/utils.py [hunk 3]
```diff
@@ -62,23 +63,26 @@ def generate_secret(length: int = 32) -> str:
 
     Returns:
         An alphanumeric string of the specified length.
+
+    Raises:
+        ValueError: If requested length is shorter than 4.
     """
     if length < 4:
-        raise ValueError("Specific secret length '%d' is too short.", length)
+        raise ValueError(f"Specific secret length '{length}' is too short.")
```
Comment (AmberCharitos) [inline src/utils.py:71]: Question: any reason you changed this back to an f-string? I though this was not the recommended approach?

File: tests/integration/conftest.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Temporal worker charm integration test config."""
```
Comment (AmberCharitos) [inline tests/integration/conftest.py:4]: suggestion: this still says `Temporal`

**Review by AmberCharitos — APPROVED** [review]
LGTM - some typos but nothing blocking

File: src/utils.py (hunk 3 above)
Comment (mertalpt) [reply src/utils.py:71]: Linter complained about lazy strings. Apparently it is lazy strings for logs and f-strings for exceptions.

Comment (AmberCharitos) [reply src/utils.py:71]: Good to know.

**Pull request merged**